network/src/primary.rs

```diff
@@ -30,32 +32,33 @@ impl Default for PrimaryNetwork {
 
         Self {
             clients: Default::default(),
+            config: Default::default(),
             retry_config,
             rng: SmallRng::from_entropy(),
         }
     }
 }
 
 impl PrimaryNetwork {
-    fn client(&mut self, address: SocketAddr) -> PrimaryToPrimaryClient<Channel> {
+    fn client(&mut self, address: Multiaddr) -> PrimaryToPrimaryClient<Channel> {
         self.clients
-            .entry(address)
-            .or_insert_with(|| Self::create_client(address))
+            .entry(address.clone())
+            .or_insert_with(|| Self::create_client(&self.config, address))
             .clone()
     }
 
-    fn create_client(address: SocketAddr) -> PrimaryToPrimaryClient<Channel> {
-        // TODO use TLS
```

Comment:
I'd like to not lose this comment, because in the mTLS context, using TLS is not just about a change of protocol, but requires configuring a custom verifier.

Reply:
Opened https://github.com/MystenLabs/mysten-infra/issues/73 for this